bool can_stop
bool on